### PR TITLE
Added make install

### PR DESCRIPTION
**Pull request by tim241**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,21 +19,44 @@ ifneq ($(MONO_OPTIONS),)
...
-	
+
+install::
+	@if [ ! -d "bin/$(CONFIGURATION)" ]; then \
+		echo "run 'make all' before you execute 'make install'!"; \
+			exit 1; \
+	fi
+	-mkdir -p "$(prefix)/lib/mono/xbuild-frameworks"
+	-mkdir -p "$(prefix)/lib/xamarin.android"
+	-mkdir -p "/usr/lib/mono/xbuild/Xamarin/"
```
Comment (jonpryor) [inline Makefile]: This shouldn't assume `/usr/lib/mono`; it should either assume `$(prefix)/lib/mono/...`, *or* it should ["probe" for mono's install prefix](https://github.com/xamarin/xamarin-android/blob/91e0ba7/tools/scripts/xabuild#L135-L147).

Comment (tim241) [inline Makefile]: Okay

**Comment by jonpryor** [comment]
This needs to update `tools/scripts/xabuild` so that it's usable when installed into `$prefix/bin`, as at present [it uses the wrong `$prefix`](https://github.com/xamarin/xamarin-android/blob/91e0ba7/tools/scripts/xabuild#L75-L76):

```bash
elif [[ "$prefix" == */bin ]] ; then
	prefix="$prefix/.."
```

For example, if you run `make install prefix=/usr/local`, the following files will be created (among many others)

* `/usr/local/bin/xabuild`
* `/usr/local/lib/xamarin.android/bin/generator`
* `/usr/local/lib/xamarin.android/lib/mandroid/class-parse.exe`
* `/usr/local/lib/xamarin.android/lib/xbuild-frameworks/MonoAndroid/v1.0/mscorlib.dll`
* `/usr/local/lib/xamarin.android/lib/xbuild/Xamarin/Android/Xamarin.Android.Common.targets`
* ...

Meanwhile, `$prefix` would be probed as `/usr/local`, which would [result in](https://github.com/xamarin/xamarin-android/blob/91e0ba7/tools/scripts/xabuild#L111-L115):

* `/p:MonoAndroidBinDirectory="/usr/local/bin"`, but *should* be `/p:MonoAndroidBinDirectory="/usr/local/lib/xamarin.android/bin"`
* `/p:MonoAndroidToolsDirectory="/usr/local/lib/mandroid"` but *should* be `/p:MonoAndroidToolsDirectory="/usr/local/lib/xamarin.android/lib/mandroid"`
* `/p:MonoDroidInstallDirectory="/usr/local"` but *should* be `/p:MonoDroidInstallDirectory="/usr/local/lib/xamarin.android"`

I'm probably missing other things as well.

You should run the `make install` and ensure that e.g. `/usr/local/bin/xabuild` can actually be used to build a project, using the appropriate directories (use `/v:diag` and read the output to see which directories are used).

File: Makefile [hunk 2]
```diff
@@ -19,9 +19,6 @@ ifneq ($(MONO_OPTIONS),)
 export MONO_OPTIONS
 endif
 
-prefix=/usr/local
```
Comment (jonpryor) [inline Makefile]: Why did you get rid of `$(prefix)`? It's still required.

File: Makefile [hunk 3]
```diff
@@ -36,18 +33,18 @@ install::
 	fi
 	-mkdir -p "$(prefix)/lib/mono/xbuild-frameworks"
 	-mkdir -p "$(prefix)/lib/xamarin.android"
-	-mkdir -p "/usr/lib/mono/xbuild/Xamarin/"
+	-mkdir -p " $(prefix)/lib/mono/xbuild/Xamarin/"
```
Comment (jonpryor) [inline Makefile]: This contains a space. You don't want it to contain a space. :-)

File: Makefile (hunk 2 above)
Comment (tim241) [inline Makefile]: Oops, let me add that again

File: Makefile (hunk 3 above)
Comment (tim241) [inline Makefile]: Sorry, copy pasting hehe :)

**Comment by tim241** [comment]
@jonpryor I already tried to compile this project with ```xabuild```: https://github.com/tim241/rootr , I also deployed it on my phone and it works, so I'll run ```xabuild Xamarin.Android-Tests.sln``` when mono is done with compiling.

**Comment by tim241** [comment]
@jonpryor Okay ```error : Tool executable '/usr/bin/../bin/generator' could not be found```, I'll need to fix ```xabuild```

File: Makefile [hunk 4]
```diff
@@ -25,15 +26,35 @@ all::
 
 all-tests::
 	MSBUILD="$(MSBUILD)" tools/scripts/xabuild $(MSBUILD_FLAGS) Xamarin.Android-Tests.sln
-	
+
+install::
+	@if [ ! -d "bin/$(CONFIGURATION)" ]; then \
+		echo "run 'make all' before you execute 'make install'!"; \
+			exit 1; \
```
Comment (jonpryor) [inline Makefile]: This is indented one too many.

**Comment by jonpryor** [comment]
build

**Pull request merged**